torch/csrc/jit/codegen/cuda/scheduler/reduction_utils.cpp

```diff
@@ -331,58 +302,11 @@ void multiReductionInliner(
       } else {
         keep_unrolled.emplace(output);
       }
-
-      // If an output has multiple consumers don't process compute at structure
-      // here, we want only terminating outputs
-      if (cached_output->uses().size() > 1) {
-        continue;
-      }
-
-      auto pos_it = std::find_if(
-          output->domain()->domain().begin(),
-          output->domain()->domain().end(),
-          [&mapped_to_trivial_reduction](IterDomain* id) {
-            return id->getParallelType() == ParallelType::Unswitch ||
-                id->getParallelType() == ParallelType::Unroll ||
-                id->getParallelType() == ParallelType::Vectorize ||
-                id->getParallelType() == ParallelType::MisalignedVectorize ||
-                mapped_to_trivial_reduction.count(id);
-          });
-      auto pos = pos_it == output->domain()->domain().end()
-          ? -1
-          : std::distance(output->domain()->domain().begin(), pos_it) + 1;
-
-      cached_output->computeAt(output, pos, ComputeAtMode::BestEffort);
-
-      compute_to.push_back(cached_output);
     }
 
-    {
-      // Add inputs to compute_at that weren't unrolled
-      auto processed_inputs = ir_utils::inputTvsOf(compute_from);
-      std::unordered_set<TensorView*> processed_inputs_set{
-          processed_inputs.begin(), processed_inputs.end()};
-      for (auto inp_tv : ir_utils::filterByType<TensorView>(fusion->inputs())) {
-        if (!processed_inputs_set.count(inp_tv)) {
-          compute_from.push_back(inp_tv);
-        }
-      }
-
-      auto processed_outputs = ir_utils::inputTvsOf(compute_to);
-      std::unordered_set<TensorView*> processed_outputs_set{
-          processed_outputs.begin(), processed_outputs.end()};
-      for (auto out_tv :
-           ir_utils::filterByType<TensorView>(fusion->outputs())) {
-        if (!processed_outputs_set.count(out_tv) && out_tv->uses().empty()) {
-          compute_to.push_back(out_tv);
-        }
-      }
-    }
-
-    // Before compute at-ing the internal structure, remove vectorization
-    // anywhere it doesn't belong. Otherwise it will mess up our inlining. Clear
-    // explicit unroll or vectorization when not for input or output GMEM
-    // transfers.
+    // Before running the inlining pass, remove vectorization anywhere it
+    // doesn't belong. Otherwise it will mess up our inlining. Clear explicit
+    // unroll or vectorization when not for input or output GMEM transfers.
     for (auto tv : ir_utils::allTvs(fusion)) {
```

Comment:
If I read correctly, we only want to unroll/vectorize the tensors inside `keep_unrolled`? If this is the case, would it be possible to replace the above
```C++
scheduler_utils::parallelizeAllLike(reference_tv);
```
with
```C++
scheduler_utils::parallelizeAllLike(reference_tv, {}, allParallelTypesExcept({ParallelType::Unroll, ParallelType::Vectorize, ParallelType::MisalignedVectorize}));
scheduler_utils::parallelizeAllLike(reference_tv, keep_unrolled, {ParallelType::Unroll, ParallelType::Vectorize, ParallelType::MisalignedVectorize});
if (keep_unrolled.count(reference_tv) == 0) {
  // cleanup reference_tv
}
```
so you don't need to cleanup all tensors here?